libraries/core_libs/consensus/src/dag/sortition_params_manager.cpp

```diff
@@ -107,7 +113,28 @@ SortitionParams SortitionParamsManager::getSortitionParams(std::optional<uint64_
   return p;
 }
 
+uint16_t calculateEfficiencyHF2(const SyncBlock& block, uint16_t stale_difficulty) {
+  auto accLambda = [&stale_difficulty](uint32_t s, const DagBlock& b) -> uint32_t {
+    if (b.getDifficulty() == stale_difficulty) {
+      return s;
+    }
+    return s + b.getTrxs().size();
+  };
+  size_t total_count = std::accumulate(block.dag_blocks.begin(), block.dag_blocks.end(), 0, accLambda);
+  uint16_t res = 100 * kOnePercent;
+  if (total_count != 0) {
+    res = block.transactions.size() * 100 * kOnePercent / total_count;
+  }
+  // this needed because stale blocks could have efficient transactions and efficiency would be bigger than 100%
+  res = std::min(uint16_t(100 * kOnePercent), res);
```

Comment:
I guess it is by design that number 10000 represents 100% ?

Reply:
Yes. It is not a good idea to use float/double for such calculations. So I defining one percent as 100. I guess that precision is enough for that calculations.